### ISSUE TITLE
Can't customize chainId when using forking configuration

### ISSUE THREAD

**Issue by frangio** [nomiclabs/hardhat#1042]
I tried the following configuration but the chain id is still 1:

```
  networks: {
    hardhat: {
      chainId: 31337,
      forking: {
        url: `https://cloudflare-eth.com/`,
      },
    },
  },
```

Is there a reason why the chain id can't be customized when forking?

**Comment by fvictorio** [comment]
I don't think so. I'm going to tentatively label this as a bug.

**Comment by fvictorio** [comment]
Besides fixing this, we shouldn't use the remote chain id when creating the forked network, since that could be potentially dangerous. We should do the same we do in the regular hardhat network: use 31337 as the chain id, and allow users to set another one with the `chainId` property.

**Comment by chanhosuh** [comment]
@fvictorio could you explain further why using the forked network's chain ID could be dangerous?  We do testing on forked networks and configurations check the chain ID to see which addresses they should load.  That seems natural to us, and I expect a lot of people do it that way and that's probably why nobody has noticed this bug until now.

**Comment by chanhosuh** [comment]
Incidentally, was this not fixed in 2.0.6, re: https://github.com/nomiclabs/hardhat/pull/1083 ?

**Comment by alcuadrado** [comment]
> could you explain further why using the forked network's chain ID could be dangerous?

@chanhosuh: You are signing txs that can be broadcasted to the forked network, accidentally or by an attacker. If you are sure this won't happen, or you are ok with that risk, you can explicitly set the same chainId.

You are right, this is now fixed, so I'm closing this issue.
